src/workspaces/common/WorkspaceSelector.tsx

```diff
@@ -0,0 +1,63 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import { CloudProviderIcon } from 'src/components/CloudProviderIcon';
+import { SelectProps, VirtualizedSelect } from 'src/components/common';
+import {
+  cloudProviderLabels,
+  getCloudProviderFromWorkspace,
+  WorkspaceWrapper as Workspace,
+} from 'src/workspaces/utils';
+
+interface WorkspaceSelectorProps
+  extends Omit<SelectProps<string, false, { value: string; label: string }>, 'value' | 'options'> {
+  workspaces: Workspace[];
+  value: string | undefined;
+  onChange: (string) => void;
+  id?: string | undefined;
+  'aria-label'?: string | undefined;
+}
+
+interface WorkspaceSelectorOption {
+  'aria-label': string;
+  value: string;
+  label: string;
+  workspace: Workspace;
+}
+
+export const WorkspaceSelector = (props: WorkspaceSelectorProps) => {
+  const { workspaces, value, onChange, id, 'aria-label': ariaLabel, ...otherProps } = props;
+  const options = _.flow(
+    _.sortBy((ws: Workspace) => ws.workspace.name.toLowerCase()),
+    _.map(({ workspace: { workspaceId, name, cloudPlatform, bucketName } }) => ({
+      'aria-label': `${cloudProviderLabels[cloudPlatform]} ${name}`,
```

Comment:
The `aria-label` being set here isn't doing anything. For one thing, `cloudProviderLabels[cloudPlatform]` is undefined because the key should be `cloudPlatform.toUpperCase()`. But then the `aria-label` is just a property of options, but `formatOptionLabel` is being used to create the HTML.

I would suggest just deleting the `aria-label` here. I played with trying to specify it in `formatOptionLabel`, but had difficulty getting into the right place in the HTML to be recognized. The workspace name is functioning as the label (for screen reader also), which means the cloud platform isn't announced, but that seems pretty minor. And what is here isn't working at all, so it's confusing.